docs/typescript/failures.md

```diff
@@ -61,6 +61,12 @@ When a Workflow or Activity fails with an unhandled error, Temporal checks if th
 
 Workflows and Activities may also throw [`ApplicationFailure.nonRetryable`](https://typescript.temporal.io/api/classes/client.applicationfailure#nonretryable-1) to expressly prevent retries.
 
+> Before TypeScript SDK v0.17.0, throwing any error in a Workflow would cause the Workflow execution to fail - in other words, all errors were "non-retryable".
+> After v0.17.0, only the Workflow _task_ fails on an error - so the Workflow will retry by default.
+> To fail the Workflow _execution_ without retries, throw `ApplicationFailure.nonRetryable`.
```

Comment:
We'll need to fix that.

The expected behavior is:
- Non retryable application failure -> fails the workflow and cannot be retried
- retryable application failure -> fails the workflow and can be retried according to the retry policy
- Other `TemporalFailure`s -> same as retryable application failure
- Any other error -> fails the workflow task and can be retried